src/moonlink/src/storage/cache/metadata/base_cache.rs

```diff
@@ -0,0 +1,49 @@
+use async_trait::async_trait;
+
+#[async_trait]
+#[allow(dead_code)]
+pub trait MetadataCacheTrait<K, V>: Send + Sync
+// TODO: Do we really need 'static here? Revisit later.
+where
+    K: Send + Sync + 'static,
+    V: Send + Sync + 'static,
+{
+    /// Retrieves a value for the given key.
+    ///
+    /// **Note:** This returns a cloned copy of the value stored in the cache.
+    /// Modifying the returned value does not affect the cached value.
+    async fn get(&self, key: &K) -> Option<V>
+    where
+        V: Clone;
+
+    /// Inserts a key-value pair into the cache.
+    ///
+    /// **Behavior:**
+    /// - If the key does not exist, the key-value pair is inserted.
+    /// - If the key already exists, the old value is overwritten with the new value.
```

Comment:
Also need to document the expiration behavior: if the key-value pair gets overriden, do we recalculate the timeout or it heherits.